internal/model/general.go

```diff
@@ -0,0 +1,6 @@
+package model
```

Comment:
I'm not a fan of `model` packages. Even though this pattern is used in the server, it's very "non go". I would much rather organize code by functionality.

What do you think about moving `plugin.go` into the `api` package (for now)?

Reply:
Agree that this is probably non-go, but given we use the `model` structs in both `api` and `store`, it seems weird to make `store` depend on `api`.

I might perhaps push for just flattening `internal` altogether, but that just kicks the problem down the road. Do you know of another project with similar fundamentals that organizes itself in a more idiomatic way?

See also https://github.com/mattermost/mattermost-cloud/tree/master/internal to which we could make an argument for Mattermost consistency (though I note I contributed this layout to both).

Reply:
> it seems weird to make `store` depend on `api`.

Good point :+1: 

> Do you know of another project with similar fundamentals that organizes itself in a more idiomatic way?

I've been searching for idiomatic package layout for a while now, but have found anything great yet.

Maybe move `plugin.go` into it's own package (`plugin`)?

Reply:
I guess that would just be renaming `model` to `plugin` at this point? I'm hesitant to make this change right now, since the benefit to deviating from `model` isn't quite clear to me yet. I remain open to discussion :)